sklearn/model_selection/_search.py

```diff
@@ -244,10 +244,11 @@ def __iter__(self):
             grid_size = len(param_grid)
 
             if grid_size < self.n_iter:
-                raise ValueError(
-                    "The total space of parameters %d is smaller "
-                    "than n_iter=%d. For exhaustive searches, use "
-                    "GridSearchCV." % (grid_size, self.n_iter))
+                warnings.warn(
+                    'The total space of parameters %d is smaller '
+                    'than n_iter=%d. For exhaustive searches, use '
+                    'GridSearchCV.' % (grid_size, self.n_iter), RuntimeWarning)
+                self.n_iter = grid_size
```

Comment:
Hmm. We usually try to avoid modifying the user-friendly provided parameters. Use a local variable instead

Reply:
So something like this?
````
if all_lists:
            # look up sampled parameter settings in parameter grid
            param_grid = ParameterGrid(self.param_distributions)
            grid_size = len(param_grid)
            iterations = self.n_iter

            if grid_size < iterations:
                warnings.warn(
                    'The total space of parameters %d is smaller '
                    'than n_iter=%d. Running %d iterations. For exhaustive searches, '
                    'use GridSearchCV.' % (grid_size, self.n_iter, grid_size),
                     RuntimeWarning)
                iterations = grid_size
            for i in sample_without_replacement(grid_size, iterations,
                                                random_state=rnd):
                yield param_grid[i]

Reply:
I will change iterations to n_iter.